config/initializers/saml_settings.rb

```diff
@@ -1,34 +1,24 @@
 module SAML
 
-  cert_file = ENV["CERTIFICATE_FILE"]
-  key_file = ENV["KEY_FILE"]
-
-  CERTFILES_EXIST = cert_file && File.exists?(cert_file) &&
-                    key_file && File.exists?(key_file)
-  NO_LOGIN_MODE = Rails.env.test? || !CERTFILES_EXIST
+  CONFIG = Rails.application.config_for(:saml).freeze
   SETTINGS = OneLogin::RubySaml::Settings.new
 
-  if NO_LOGIN_MODE
-    msg = "No SAML settings configured, starting up in NO LOGIN MODE."\
-          " ID.me integration will not function."
-    Rails.logger.warn msg
-  else
+  unless Rails.env.test?
+    Figaro.require_keys("CERTIFICATE_FILE", "KEY_FILE")
```

Comment:
have not looked at this PR, but saw @jkassemi put in this issue https://github.com/department-of-veterans-affairs/devops/issues/430 which feels relevant to the secrets.yml conversation above.  Just connecting dots...